### PR TITLE
Natural Scene Components

### PR DESCRIPTION
**Pull request by Rheeseyb**
**Problem:**
We want to replace `<Scene component={App} />` with `<Scene><App/></Scene>`

**Fix:**
This is the final step of that, but I'm splitting into 2 PRs - one introducing the changes to support natural scenes, and then a second upcoming PR to remove the old style scenes (for the sake of keeping the functional change PR as free from noise as possible).

This PR introduces a very basic `Scene` component into `utopia-api` (so that user's projects can build and run outside of Utopia), and a `SceneComponent` component that wraps the above when running in the canvas (to provide styling based on the canvas state). It also updates the targeting functionality used throughout for selection and insertion to reflect the updated hierarchy, and also updates the indentation depth in the Navigator for the same reason.

**Demo:**
natural-scenes

**Note:**
This work hasn't ported across the dynamic / static scene functionality - I'm not sure if that still makes sense now... (will bring that up in Slack)

### CODE REVIEW

**Review by balazsbajorics — APPROVED** [review]

**Comment by maltenuhn** [comment]
> This work hasn't ported across the dynamic / static scene functionality

Isn't that basically just the layout section for normal elements at this point, since autosizing or full and partial fixed sizing would be accomplished with standard CSS?

```
FixedSizeComponent = props => <div style={{ width: 200, height: 300 }}>{props.children}</div>
VarSizeComponent = props => <div style={{ padding: 40, border: '4px solid mauve' }}>{props.children}</div>

/* Scenes 'hug content'  and resize with it */
<Scene alt='first'>
  <FixedSizeComponent>Hi</FixedSizeComponent>
</Scene>
<Scene alt='second'>
  <VarSizeComponent>Hi</VarSizeComponent>
</Scene>

/* Scene is fixed in one dimension */
<Scene alt='third' style={{ minWidth: 300 }} >
  <VarSizeComponent>Hi</VarSizeComponent>
</Scene>
```

etc etc

**Comment by balazsbajorics** [comment]
> > This work hasn't ported across the dynamic / static scene functionality
> 
> Isn't that basically just the layout section for normal elements at this point, since autosizing or full and partial fixed sizing would be accomplished with standard CSS?
> 
> ```
> FixedSizeComponent = props => <div style={{ width: 200, height: 300 }}>{props.children}</div>
> VarSizeComponent = props => <div style={{ padding: 40, border: '4px solid mauve' }}>{props.children}</div>
> 
> /* Scenes 'hug content'  and resize with it */
> <Scene alt='first'>
>   <FixedSizeComponent>Hi</FixedSizeComponent>
> </Scene>
> <Scene alt='second'>
>   <VarSizeComponent>Hi</VarSizeComponent>
> </Scene>
> 
> /* Scene is fixed in one dimension */
> <Scene alt='third' style={{ minWidth: 300 }} >
>   <VarSizeComponent>Hi</VarSizeComponent>
> </Scene>
> ```
> 
> etc etc

there's definitely active and special-cased logic when you _resize_ a dynamic/static scene. the update function figures out if it should change the props of the scene, or change the props of the root element. That logic is not generic enough to be able to say "this should now run for every flow div resize"

**Comment by maltenuhn** [comment]
> there's definitely active and special-cased logic when you resize a dynamic/static scene. 

Ah ok, I thought about it only in terms of switching between auto-sizing and giving a scene dimensions.

The "I can resize a scene and it resizes the children" behaviour was quite useful when almost all components (courtesy of "View" had fixed dimensions), but with Scenes being more like mini-browsers I think it can go.


 times where almost every component had fixed dimensions by default, courtesy of `View`, but today I'd miss it a lot less. I'd say

File: editor/src/components/canvas/controls/select-mode-control-container.tsx [hunk 1]
```diff
@@ -494,8 +496,14 @@ export class SelectModeControlContainer extends React.Component<
   render() {
     const cmdPressed = this.props.keysPressed['cmd'] || false
     const allElementsDirectlySelectable = cmdPressed && !this.props.isDragging
-    const roots = MetadataUtils.getAllStoryboardChildrenPathsScenesOnly(
-      this.props.componentMetadata,
+    const storyboardChildren = MetadataUtils.getAllStoryboardChildren(this.props.componentMetadata)
+    const roots = mapDropNulls(
```
Comment (seanparsons) [inline editor/src/components/canvas/controls/select-mode-control-container.tsx:500]: Should the logic here and in `insert-mode-control-container.tsx` be refactored out as they look identical?

File: editor/src/components/navigator/navigator-item/navigator-item.tsx [hunk 2]
```diff
@@ -32,13 +32,7 @@ interface ComputedLook {
...
 
 export function getElementPadding(templatePath: TemplatePath): number {
-  // if an element is the child of the Storyboard component, let's show it at the root level
-  const extraDepthRemoval = TP.pathsEqual(
-    TP.scenePathPartOfTemplatePath(templatePath),
-    EmptyScenePathForStoryboard,
-  )
-  const depthOffset = extraDepthRemoval ? 2 : 0
-  return (TP.navigatorDepth(templatePath) - 1 - depthOffset) * BasePaddingUnit
+  return TP.navigatorDepth(templatePath) * BasePaddingUnit
```
Comment (seanparsons) [inline editor/src/components/navigator/navigator-item/navigator-item.tsx:35]: Nice!

File: editor/src/core/model/element-metadata-utils.ts [hunk 3]
```diff
@@ -769,13 +771,22 @@ export const MetadataUtils = {
...
+    // FIXME Replace with a property controls check
+    const elementEither = instance.element
+
+    if (isLeft(elementEither)) {
+      return intrinsicHTMLElementNamesThatSupportChildren.includes(elementEither.value)
     } else {
-      return true
+      const element = elementEither.value
+      if (isJSXElement(element) && isUtopiaAPIComponent(element.name, imports)) {
+        // Explicitly prevent components / elements that we *know* don't support children
```
Comment (seanparsons) [inline editor/src/core/model/element-metadata-utils.ts:782]: This comment now appears out of date compared to the code it's above.

**Review by seanparsons — APPROVED** [review]

**Pull request merged**